src/AWSCredentials.jl

```diff
@@ -56,392 +60,497 @@ mutable struct AWSCredentials
 
-    # Define our search options
+    # Define our search options, expected to be callable with no arguments. Should return
+    # `nothing` when credentials are not able to be located
     functions = [
-        env_instance_credentials,
+        get_awscreds_env_vars,
         () -> dot_aws_credentials(profile),
         () -> dot_aws_config(profile),
-        instance_credentials,
+        _get_awscreds_instance,
     ]
 
     # Loop through our search locations until we get credentials back
     for f in functions
-        renew = f
-        creds = renew()
+        credential_function = f
+        creds = credential_function()
         creds === nothing || break
     end
 
     creds === nothing && error("Can't find AWS credentials!")
-    creds.renew = renew
-
-    if debug_level > 0
-        display(creds)
-        println()
-    end
+    creds.renew = credential_function
 
     return creds
 end
 
-will_expire(cr::AWSCredentials) = now(UTC) >= cr.expiry - Minute(5)
-
-"""
-    check_credentials(cr::AWSCredentials; force_refresh::Bool=false)
-
-Checks current AWSCredentials, refreshing them if they are soon to expire.
-If force_refresh is `true` the credentials will be renewed immediately.
-"""
-function check_credentials(cr::AWSCredentials; force_refresh::Bool=false)
-    if force_refresh || will_expire(cr)
-        if debug_level > 0
-            println("Renewing credentials... ")
-        end
-        renew = cr.renew
-
-        if renew !== nothing
-            new_creds = renew()
-
-            new_creds === nothing && error("Can't find AWS credentials!")
-            copyto!(cr, new_creds)
-
-            # Ensure renewal function is not overwritten by the new credentials
-            cr.renew = renew
-        else
-            if debug_level > 0
-                println("Credentials cannot be renewed...")
-            end
-        end
-    end
-
-    return cr
-end
-
-function Base.show(io::IO,c::AWSCredentials)
-    println(io, string(c.user_arn,
-                       c.user_arn == "" ? "" : " ",
-                       "(",
-                       c.account_number,
-                       c.account_number == "" ? "" : ", ",
-                       c.access_key_id,
-                       c.secret_key == "" ? "" : ", $(c.secret_key[1:3])...",
-                       c.token == "" ? "" : ", $(c.token[1:3])..."),
-                       ")")
-end
-
-function Base.copyto!(dest::AWSCredentials, src::AWSCredentials)
-    for f in fieldnames(typeof(dest))
-        setfield!(dest, f, getfield(src, f))
-    end
-end
-import Base: copy!
-Base.@deprecate copy!(dest::AWSCredentials, src::AWSCredentials) copyto!(dest, src)
-
 
 """
-Is Julia running in an AWS Lambda sandbox?
-"""
-localhost_is_lambda() = haskey(ENV, "LAMBDA_TASK_ROOT")
+    localhost_is_ec2()
 
+Checks to see if the localhost is on EC2.
 
-"""
-Is Julia running on an EC2 virtual machine?
+# Returns
+- `Bool`: True if running the machine executing this is an EC2 instance
```

Comment:
Remove this return section as we can be descriptive enough that we don't need it.